src/routes/view-content/+page.svelte

```diff
@@ -50,17 +50,24 @@
     import BibleMenu from './bible-menu/BibleMenu.svelte';
     import BookPassageSelectorPane from './bible-menu/BookPassageSelectorPane.svelte';
     import BookChapterSelectorPane from './bible-menu/BookChapterSelectorPane.svelte';
-    import type { ApiParentResource } from '$lib/types/resource';
+    import {
+        MediaType,
+        type ApiParentResource,
+        type ResourceContentInfoWithMetadata,
+        type TextResourceContentJustId,
+    } from '$lib/types/resource';
     import { selectedBibleSection } from '$lib/stores/passage-form.store';
     import SettingsMenu from './settings-menu/SettingsMenu.svelte';
     import type { BibleSection } from '$lib/types/bible';
     import BibleUnavailable from './BibleUnavailable.svelte';
     import { bibleSectionToReference } from '$lib/utils/bible-section-helpers';
     import QuickShare from './quick-share/QuickShare.svelte';
     import GuideContent from './guide-content/GuideContent.svelte';
+    import FullscreenTextResource from './library-resource-menu/FullscreenTextResource.svelte';
 
     let bibleData: BibleData | null = null;
     let resourceData: ResourceData | null = null;
+    let fullscreenTextResourceStack: (ResourceContentInfoWithMetadata | TextResourceContentJustId)[] = [];
```

Comment:
Using a stack to hold the list of fullscreen text resources. This lets the user tap on a trail of links and get back to where they were.